src/botPage/view/blockly/index.js

```diff
@@ -266,20 +266,35 @@ export default class _Blockly {
                     const defaultStrat = parseQueryString().strategy;
                     const xmlFile = defaultStrat ? `xml/${defaultStrat}.xml` : 'xml/main.xml';
 
-                    $.get(xmlFile, main => {
-                        repaintDefaultColours();
-                        overrideBlocklyDefaultShape();
-                        this.blocksXmlStr = Blockly.Xml.domToPrettyText(main);
-                        Blockly.Xml.domToWorkspace(main.getElementsByTagName('xml')[0], workspace);
-                        this.zoomOnPlusMinus();
-                        disposeBlocksWithLoaders();
-                        setTimeout(() => {
-                            setBeforeUnload(true);
-                            Blockly.mainWorkspace.cleanUp();
-                            Blockly.mainWorkspace.clearUndo();
-                        }, 0);
-                        resolve();
-                    });
+                    const getFile = xml => {
+                        $.get(xml, main => {
+                            repaintDefaultColours();
+                            overrideBlocklyDefaultShape();
+                            this.blocksXmlStr = Blockly.Xml.domToPrettyText(main);
+                            Blockly.Xml.domToWorkspace(main.getElementsByTagName('xml')[0], workspace);
+                            this.zoomOnPlusMinus();
+                            disposeBlocksWithLoaders();
+                            setTimeout(() => {
+                                setBeforeUnload(true);
+                                Blockly.mainWorkspace.cleanUp();
+                                Blockly.mainWorkspace.clearUndo();
+                            }, 0);
+
+                            localStorage.setItem('previousStrat', xml);
+                            resolve();
+                        }).catch(() => {
+                            const previousStrat = localStorage.getItem('previousStrat') || 'xml/main.xml';
+                            getFile(previousStrat);
+
+                            globalObserver.emit('Notify', {
+                                className: 'warn',
+                                message  : translate('Invalid link'),
```

Comment:
Could you change the message into something like _"The strategy you tried to load is invalid."_?